homeassistant/components/mikrotik/device.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, mac: str, params: dict[str, Any]) -> None:
     @property
     def name(self) -> str:
         """Return device name."""
-        return self._params.get("host-name", self.mac)
+        return str(self._params.get("host-name", self.mac))
```

Comment:
Do we need to string copy? Or would casting be enough?

Reply:
I think casting is enough as the device name should always be string even if the returned value from the library is an integer.

Reply:
In that case, it should be casted, as that operating is a no-op in production.

Reply:
Sorry but I am not sure what you mean. Can you please suggest the solution.

Reply:
```suggestion
        return cast(str, self._params.get("host-name", self.mac))
```

Reply:
I tried this but got an error when running the tests "TypeError: decoding to str: need a bytes-like object, int found". Just to clarify this required in case the "host-name" key is an integer (like {"host-name": 123}). In this case we want the device `name` to be `"123"` instead of `123`

Reply:
Aah ok, in that case it is correct 👍 
Sorry about that.